### ISSUE TITLE
Infrastructure: Profiling Setup

### ISSUE THREAD

**Issue by Goldziher** [starlite-api/starlite#416]
The problem we need to address is how to profile Starlite to identify bottlenecks. We previously used benchmarking of functions, and benchmarking of request/response cycles, but these are not particularly efficent. 

It would be therefore great to have some way to  profile the code base. 

One idea for example is to create a script that launches our example project, seeds the db and in a subshell fires requests intended to activate different code paths. 

This though is just am idea, any workable solution is worth discussion.

**Comment by peterschutt** [comment]
Maybe we should try to emulate the techempower benchmarking?

**Comment by Goldziher** [comment]
> Maybe we should try to emulate the techempower benchmarking?

Can you explain?

**Comment by MatthijsKok** [comment]
The [TechEmpower Framework Benchmarks](https://github.com/TechEmpower/FrameworkBenchmarks) describes itself as:
> This project provides representative performance measures across a wide field of web application frameworks. With much help from the community, coverage is quite broad and we are happy to broaden it further with contributions. The project presently includes frameworks on many languages including Go, Python, Java, Ruby, PHP, C#, F#,Clojure, Groovy, Dart, JavaScript, Erlang, Haskell, Scala, Perl, Lua, C, and others. The current tests exercise plaintext responses, JSON serialization, database reads and writes via the object-relational mapper (ORM), collections, sorting, server-side templates, and XSS counter-measures. Future tests will exercise other components and greater computation.  

TL;DR It's a high-quality unbiased benchmark suite that almost all large frameworks use to compare with eachother.  

[Benchmark results](http://www.techempower.com/benchmarks/)

In my opinion it would be the best way to show the power of the routing radix tree over regex based matching.

**Comment by Goldziher** [comment]
> The [TechEmpower Framework Benchmarks](https://github.com/TechEmpower/FrameworkBenchmarks) describes itself as:
> 
> > This project provides representative performance measures across a wide field of web application frameworks. With much help from the community, coverage is quite broad and we are happy to broaden it further with contributions. The project presently includes frameworks on many languages including Go, Python, Java, Ruby, PHP, C#, F#,Clojure, Groovy, Dart, JavaScript, Erlang, Haskell, Scala, Perl, Lua, C, and others. The current tests exercise plaintext responses, JSON serialization, database reads and writes via the object-relational mapper (ORM), collections, sorting, server-side templates, and XSS counter-measures. Future tests will exercise other components and greater computation.
> 
> TL;DR It's a high-quality unbiased benchmark suite that almost all large frameworks use to compare with eachother.
> 
> [Benchmark results](http://www.techempower.com/benchmarks/)
> 
> In my opinion it would be the best way to show the power of the routing radix tree over regex based matching.

Thanks. We are all aware of what this is :). The question was to understand what @peterschutt meant by his suggestion, since whats required in this issue is a profiling setup - which isnt this exactly. 

Regarding tech powered, there are some issues there. Our benchmarks are not running correctly there, as well as starlette's for some reason. @peterschutt started exploring this, but its not clear whats going on there exactly.
